m4/mpi.m4

```diff
@@ -3,19 +3,44 @@ dnl check for the required MPI library
 dnl ---------------------------------------------------------------------------
 AC_DEFUN([ACX_MPI], [
 
-dnl if MPIHOME is empty, set it to /usr
-AS_IF([test "x$MPIHOME" = "x"], [MPIHOME="/usr"])
+dnl if MPIHOME is empty, set it to an invalid directory and try to find the correct location.
+AS_IF([test "x$MPIHOME" = "x"], [MPIHOME="/.."])
 
 AC_ARG_WITH([mpi],
             AS_HELP_STRING([--with-mpi=PATH],
                            [Prefix where MPI is installed (MPIHOME)]),
             [MPI="$withval"],
             [
-              AS_ECHO(["note: MPI library path not given... trying prefix=$MPIHOME"])
+              AS_ECHO(["note: MPI library path not given... trying to find it myself"])
               MPI=$MPIHOME
             ])
 
-AS_IF([test -z "$MPI"], [MPI="/usr"])
+AS_IF([test -z "$MPI"], [MPI="/.."])
+
+dnl if MPI is not set in the config, search it via the $INCLUDE and $LIBRARY_PATH variables;
+dnl     with this we can hope to get a configure that is consistent with the environment.
+AS_IF([test "$MPI" = "/.."],
+ [
+    for i in $(echo $LIBRARY_PATH | tr ':' '\n')
```

Comment:
According to this [SO question](https://stackoverflow.com/questions/4250624/ld-library-path-vs-library-path), `$LIBRARY_PATH` is searched by gcc/g++ when linking. Anyone know if that's true for other compilers as well? I don't believe LIBRARY_PATH is something I've ever used on any system...

Reply:
This is a good point; so we probably should do this step only if the gcc is used... 

I have just seen that according to the configure-script, libraries can be also given via
```
  CXXFLAGS    C++ compiler flags
  LDFLAGS     linker flags, e.g. -L<lib dir> 
  LIBS        libraries to pass to the linker, e.g. -l<library>
  CPPFLAGS    (Objective) C/C++ preprocessor flags, e.g. -I<include dir>
  CFLAGS      C compiler flags
  libmesh_CPPFLAGS      User-specified C/C++ preprocessor flags
  libmesh_CXXFLAGS      User-specified C++ compilation flags
  libmesh_CFLAGS           User-specified C compilation flags
```
So we should check these variables as well, shouldn't we?